handlers_test.go

```diff
@@ -50,42 +50,91 @@ func TestBadRequest(t *testing.T) {
 		// missing request body
 		{`/sign/data`, `POST`, ``},
 		{`/sign/hash`, `POST`, ``},
+		{`/sign/file`, `POST`, ``},
+		{`/sign/files`, `POST`, ``},
 		// invalid json body
 		{`/sign/data`, `POST`, `{|||...........`},
 		{`/sign/hash`, `POST`, `{|||...........`},
+		{`/sign/file`, `POST`, `{|||...........`},
+		{`/sign/files`, `POST`, `{|||...........`},
 		// missing input
-		{`/sign/data`, `POST`, `[{"input": "", "keyid": "abcd"}]`},
-		{`/sign/hash`, `POST`, `[{"input": "", "keyid": "abcd"}]`},
+		{`/sign/data`, `POST`, `[{"input": ""}]`},
+		{`/sign/hash`, `POST`, `[{"input": ""}]`},
+		{`/sign/file`, `POST`, `[{"input": ""}]`},
+		{`/sign/files`, `POST`, `[{"input": ""}]`},
 		// input not in base64
 		{`/sign/data`, `POST`, `[{"input": "......."}]`},
 		{`/sign/hash`, `POST`, `[{"input": "......."}]`},
-		// asking for a xpi signature using a hash will fail
+		{`/sign/file`, `POST`, `[{"input": "......."}]`},
+		{`/sign/files`, `POST`, `[{"input": "......."}]`},
+
+		// missing files
```

Comment:
Tests look great! Is there one for the `unsigned filename must match ^[-_\.a-zA-Z0-9]{1,256}$` check that I'm missing?

Reply:
Not sure if it's needed, but would it be worth including a test for a filename with say `@` or some other non-valid character in it?

Reply:
Good call! I added a few more in d99e8754.

Checked the OWASP cheat sheet too: https://cheatsheetseries.owasp.org/cheatsheets/File_Upload_Cheat_Sheet.html#filename-sanitization